### ISSUE TITLE
using pytorch training reference script with pytorch-metric-learning

### ISSUE THREAD

**Issue by abaqusstudent** [KevinMusgrave/pytorch-metric-learning#505]
Hi @KevinMusgrave !

I have recently been using the reference scripts pytorch provides to train my models (which is wonderful btw) BUT, I would love to use pytorch-metric-learning with this reference script.

The training script and blog re this is here:
https://github.com/pytorch/vision/tree/main/references/classification
https://pytorch.org/blog/how-to-train-state-of-the-art-models-using-torchvision-latest-primitives/

I am particularly interested in classification:
https://github.com/pytorch/vision/blob/main/references/classification/train.py

However, the issue ofc is that they all use CE loss which requires the logits - but I am not sure how to use say ArcFace Loss with this training reference script. Essentially, I would need the logits to make this work, but atm, all my arcface loss models work with an embedder output and distance metrics.

I was wondering if you could provide some guidance/advise on how to proceed to include metric learning in the reference training script.

Thank you!

**Comment by KevinMusgrave** [comment]
You'll have to modify the model to return embeddings instead of logits. Then during testing, if you want logits instead of embeddings, you can do `logits = loss_fn.get_logits(embeddings)`. This is assuming you're using a classification loss like ArcFaceLoss.

**Comment by abaqusstudent** [comment]
thank you for the fast reply @KevinMusgrave I will make the mods as you have suggested.
